src/com/sap/piper/tools/JavaArchiveDescriptor.groovy

```diff
@@ -0,0 +1,62 @@
+package com.sap.piper.tools
+
+import com.sap.piper.EnvironmentUtils
+import com.sap.piper.FileUtils
+
+import hudson.AbortException
+
+
+class JavaArchiveDescriptor extends ToolDescriptor {
+
+    final javaTool
+    final javaOptions
+
+    JavaArchiveDescriptor(name, environmentKey, stepConfigurationKey, executablePath, executableName, version, versionOption, javaTool, javaOptions) {
+        super(name, environmentKey, stepConfigurationKey, executablePath, executableName, version, versionOption)
+        this.javaTool = javaTool
+        this.javaOptions = javaOptions
+    }
+
+    @Override
+    def getHome(script, configuration, log = true) {
+
+        def home
```

Comment:
Here in this case it is not about getting a ```HOME``` directory, like ```JAVA_HOME```, ```NEO_HOME```, ```MVN_HOME```, etc. Here it is about getting the location of a jar file which is expected to be an uber jar, working with ```java -jar```.  Name ```getHome``` is misleading here.

It is more a ```getToolJar```. The home of the executable would be more ```JAVA_HOME``` in this case (```javaTool.getHome()```, maybe throwing an unsupportedOperationException would also be an option, other option: introduce a base class without ```getHome``` and without ```getToolJar``` for common params like version and inherit from that base an ExecutableDescriptor and a JarFileDescriptor, maybe we find better names).

But in case other colleagues thinks this is getting some kind of a academical discussion we can also keep it...

Reply:
I can change it to getToolLocation, but for one name changing the design is too much.